libs/atqr/domain/src/lib/contact/contact.valueObject.ts

```diff
@@ -0,0 +1,38 @@
+import { MaxLength, MinLength } from 'class-validator';
+import { EmailAddress } from '../EmailService';
+import { ValueObject } from '../../utils/valueObject';
+
+export enum ParticipationRole {
+  Contender,
+  Judge,
+}
+export class Contact extends ValueObject<Contact> {
+  @MinLength(5)
+  @MaxLength(20)
+  name: string;
+  email: EmailAddress;
+  role: ParticipationRole;
+
+  constructor(name: string, email: EmailAddress, role: ParticipationRole) {
+    super(Contact, ['name', 'email', 'role']);
+    this.name = name
+    this.email = email
+    this.role = role
+  }
+
+  public setName(newName: string): Contact {
+    return this.newInstanceWith({
+      name: newName,
```

Comment:
If you make the name of the parameter the same as the name of the property you can wite a shorter object. In this case, if you rename `newName` to `name`, you can simply write: 

`this.newInstanceWith({ name });`

And it will be the same as writing

`this.newInstanceWith({ name: name });`